pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -215,9 +222,39 @@ func (m *kubeGenericRuntimeManager) calculateLinuxResources(cpuRequest, cpuLimit
 		resources.CpuPeriod = cpuPeriod
 	}
 
+	if supportsMemoryOOMGroup() {
+		// Turn on memory.oom.group which tells the kernel to treat the entire container cgroup as
+		// a single unit for OOM kill purposes.  This will cause all processes in the container to be
+		// OOM killed together if any are OOM killed.
+		resources.Unified = map[string]string{
+			"memory.oom.group": "1",
+		}
+	}
 	return &resources
 }
 
+// supportsMemoryOOMGroup returns true if
+func supportsMemoryOOMGroup() bool {
+	supportsMemoryOOMGroupOnce.Do(func() {
+		kernelVersionStr, err := ipvs.NewLinuxKernelHandler().GetKernelVersion()
```

Comment:
is there a good place to reuse similar from sysctl (pkg/kubelet/sysctl/safe_sysctls.go)? Have you copied from there?

Also interesting that the `NewLinuxKernelHandler` in pkg/proxy/ipvs/proxier.go is not compiling only for linux like this file. I remember in sysctl we had problems trying to call it from Windows

Reply:
I think you could just assume the functionality is present if it is running on cgroup v2, since we are already using features (like the hugetlb controller) that were added much later

Reply:
It's a cgroup file, not a sysctl value.  But assuming it works if cgroupsv2 is in used SGTM, the docs say that requires kernel 5.8 or later.